src/App.jsx

```diff
@@ -1,25 +1,63 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import "./App.css";
 
-function City(props) {
-  return <div>This is the City component</div>;
+const api = "https://ctp-zip-code-api.onrender.com/";
+
+async function getZipCode(zipcode) {
+  const response = await fetch(api + "zip/" + zipcode);
+  const data = await response.json();
+  return data;
 }
 
-function ZipSearchField(props) {
-  return <div>This is the ZipSearchField component</div>;
+function City({ cityData }) {
+  if (!cityData) return "No data found";
+
+  return (
+    <div>
+      <h2>{cityData.City}</h2>
+      <p>State: {cityData.State}</p>
+      <p>Location: ({cityData.Lat}, {cityData.Long})</p>
+      <p>Population (estimated): {cityData.EstimatedPopulation}</p>
+      <p>Total Wages: {cityData.TotalWages}</p>
+    </div>
+  );
+}
+
+function ZipSearchField({ onChange }) {
+  return (
+    <input
+      type="text"
+      className="form-control"
+      placeholder="Type a zip code"
+      onChange={(e) => onChange(e.target.value)}
+    />
+  );
 }
 
 function App() {
+  const [zipCode, setZipCode] = useState("");
+  const [cityData, setCityData] = useState([]);
+
+  useEffect(() => {
+    if (zipCode.length === 5) {
+      (async () => {    //why do i need to await it again?
```

Comment:
We use `await` when we use `Promise`s. Since `getZipCode` is `async` because it uses `fetch` (which returns a Promise), we need `await`.